app/models/chargeback_container_project.rb

```diff
@@ -38,16 +38,20 @@ def self.build_results_for_report_ChargebackContainerProject(options)
     # Find Project by id or get all projects
     provider_id = options[:provider_id]
     id = options[:entity_id]
-    raise "must provide option :entity_id and provider_id" if id.nil? && provider_id.nil?
+    tag = options[:tag]
+    raise "must provide option :entity_id, provider_id or tag" if id.nil? && provider_id.nil? && tag.nil?
 
-    @groups = if provider_id == "all"
+    @groups = if tag.present?
+                # Get all groups belonging to projects with this tag
+                project_ids = ContainerProject.find_tagged_with(:all => tag, :ns => "*").select(:id)
+                ContainerGroup.where(:container_project_id => project_ids).or(ContainerGroup.where(:old_container_project_id => project_ids))
+              elsif provider_id == "all"
                 ContainerGroup.all
               elsif id == "all"
                 ContainerGroup.where('ems_id = ? or old_ems_id = ?', provider_id, provider_id)
               else
                 ContainerGroup.where('container_project_id = ? or old_container_project_id = ?', id, id)
               end
-
```

Comment:
@zeari I find it more readable with the space and I don't think it's worth having this change in the PR.